store/sqlstore/tokens_store.go

```diff
@@ -32,16 +32,15 @@ func NewSqlTokenStore(sqlStore SqlStore) store.TokenStore {
 func (s SqlTokenStore) CreateIndexesIfNotExists() {
 }
 
-func (s SqlTokenStore) Save(token *model.Token) store.StoreChannel {
-	return store.Do(func(result *store.StoreResult) {
-		if result.Err = token.IsValid(); result.Err != nil {
-			return
-		}
+func (s SqlTokenStore) Save(token *model.Token) *model.AppError {
+	if err := token.IsValid(); err != nil {
+		return err
+	}
 
-		if err := s.GetMaster().Insert(token); err != nil {
-			result.Err = model.NewAppError("SqlTokenStore.Save", "store.sql_recover.save.app_error", nil, "", http.StatusInternalServerError)
-		}
-	})
+	if err := s.GetMaster().Insert(token); err != nil {
+		return model.NewAppError("SqlTokenStore.Save", "store.sql_recover.save.app_error", nil, "", http.StatusInternalServerError)
```

Comment:
Is the "" on purpose to not leak error details, or should `err.Message` be passed there?

Reply:
I'm not 100% sure, but because can be something risky, lets keep as was before.